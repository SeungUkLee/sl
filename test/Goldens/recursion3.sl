(* Recursion *)

let rec foo = fun f g n ->
  if n == 0
    then 0
    else foo (f g) (g f) (n - 1)
in
  foo (fun x -> x) (fun y -> y) 3

(* result : type error *)
