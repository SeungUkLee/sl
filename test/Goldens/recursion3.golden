[Error] "f" expression has type "l" but an expression was expected of type "m -> l"
  expected: m -> l
    actual: l
